lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -277,3 +292,11 @@ export async function generateLockFile(
   }
   return { lockFile };
 }
+async function isCorepack(lockFileDir: string): Promise<boolean> {
+  const pkgJson = await readLocalFile(
+    upath.join(lockFileDir, 'package.json'),
+    'utf8'
+  );
+
+  return is.nonEmptyStringAndNotWhitespace(JSON.parse(pkgJson)?.packageManager);
+}
```

Comment:
@rarkins I'm thinking if it makes sense to do this at extrsaction time and add to manager config object, so we don't need to read the package file again 🤔

Reply:
Yeah, anything which doesn't change could be done at either stage

Reply:
we don't read packagel.json for other reasons though in post-update? I thought we did

Reply:
nope, we don't 🤷‍♂️ 

fixed by moving to extraction phase.